src/wazuh_db/wdb_integrity.c

```diff
@@ -638,3 +639,52 @@ int wdbi_check_sync_status(wdb_t *wdb, wdb_component_t component) {
     cJSON_Delete(j_sync_info);
     return result;
 }
+
+int wdb_get_global_group_hash(wdb_t * wdb, os_sha1 hexdigest) {
+    if (OS_SUCCESS == wdb_global_group_hash_cache_operations("read", hexdigest)) {
+        mdebug2("A global group hash was found in cache");
+        return OS_SUCCESS;
+    } else {
+        if(!wdb) {
+            mdebug1("Missing DB pointer to calculate the global group hash.");
+            return OS_INVALID;
+        }
+
+        sqlite3_stmt* stmt = wdb_init_stmt_in_cache(wdb, WDB_STMT_GLOBAL_GROUP_HASH_GET);
+        if (!stmt) {
+            return OS_INVALID;
+        }
+
+        int result = wdb_calculate_stmt_checksum(wdb, stmt, WDB_GENERIC_COMPONENT, hexdigest);
+        if(result == 1) {
+            wdb_global_group_hash_cache_operations("set", hexdigest);
+            mdebug2("New global group hash calculated and stored in cache.");
+            return OS_SUCCESS;
+        } else {
+            mdebug2("No group hash was found to calculate the global group hash.");
+            return OS_INVALID;
```

Comment:
Just wondering whether this is semantically correct. Cause in fact, the method succeded on its operations but there is no data in the database. In such case, it succeeds but returns empty data. Opinions?

Reply:
It's interesting because the lack of a non-NULL group hash is a valid situation; but I'm returning `OS_INVALID` because the **hexdigest** variable hasn't been written and I didn't want it to be read. What do you think?

Reply:
Probably we could keep it as it is implemented right now. The thing is that probably is possible to implement that change that I'm suggesting, but this would involve an extra check in the caller. In this way, we avoid the extra check and simplify the function usage.